### PR TITLE
Move basic SQL query examples to user guide

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Fixes https://github.com/apache/datafusion/issues/11210
Part of #11172 and #1813

## Rationale for this change
Hopefully this fixes the CI failure we were seeing in https://github.com/apache/datafusion/issues/11210 while also improving the documentation

There are several places in the existing documentation for running SQL via SessionContext
1. https://datafusion.apache.org/user-guide/example-usage.html#run-a-sql-query-against-data-stored-in-a-csv
2. https://docs.rs/datafusion/latest/datafusion/index.html#sql

Thus we don't need another one in the examples directory

## What changes are included in this PR?

1. Add  a SQL section in the library user guide with the basic examples (thanks @tshauck for starting this)
2. Consolidate parquet_sql, avro_sql, and csv_sql examples into the docs
3. Make sure the examples run as part of the doctests (`cargo doc ...`)
4. 

## Are these changes tested?
Yes, they are run via doctests

## Are there any user-facing changes?

New docs

### CODE REVIEW

File: datafusion/core/src/lib.rs [hunk 1]
```diff
@@ -592,3 +592,9 @@ doc_comment::doctest!(
     "../../../docs/source/user-guide/example-usage.md",
     user_guid_example_tests
 );
+
+#[cfg(doctest)]
```
Comment (alamb) [inline datafusion/core/src/lib.rs:596]: This means these tests are run as part of the doc tests

```shell
$ cargo test --doc --features avro,json -- library_user_guide

...
    Finished `test` profile [unoptimized + debuginfo] target(s) in 0.16s
   Doc-tests datafusion

running 5 tests
test datafusion/core/src/lib.rs - library_user_guide_example_usage (line 764) ... ok
test datafusion/core/src/lib.rs - library_user_guide_example_usage (line 718) ... ok
test datafusion/core/src/lib.rs - library_user_guide_example_usage (line 673) ... ok
test datafusion/core/src/lib.rs - library_user_guide_example_usage (line 775) ... ok
test datafusion/core/src/lib.rs - library_user_guide_example_usage (line 641) ... ok
...
```

File: datafusion-examples/README.md [hunk 2]
```diff
@@ -66,12 +64,11 @@ cargo run --example csv_sql
 - [`memtable.rs`](examples/memtable.rs): Create an query data in memory using SQL and `RecordBatch`es
 - [`optimizer_rule.rs`](examples/optimizer_rule.rs): Use a custom OptimizerRule to replace certain predicates
 - [`parquet_index.rs`](examples/parquet_index.rs): Create an secondary index over several parquet files and use it to speed up queries
-- [`parquet_sql.rs`](examples/parquet_sql.rs): Build and run a query plan from a SQL statement against a local Parquet file
 - [`parquet_sql_multiple_files.rs`](examples/parquet_sql_multiple_files.rs): Build and run a query plan from a SQL statement against multiple local Parquet files
 - [`parquet_exec_visitor.rs`](examples/parquet_exec_visitor.rs): Extract statistics by visiting an ExecutionPlan after execution
 - [`parse_sql_expr.rs`](examples/parse_sql_expr.rs): Parse SQL text into Datafusion `Expr`.
 - [`plan_to_sql.rs`](examples/plan_to_sql.rs): Generate SQL from Datafusion `Expr` and `LogicalPlan`
-- [`pruning.rs`](examples/parquet_sql.rs): Use pruning to rule out files based on statistics
+- [`pruning.rs`](examples/pruning.rs): Use pruning to rule out files based on statistics
```
Comment (alamb) [inline datafusion-examples/README.md:71]: drive by cleanup

File: datafusion-examples/README.md [hunk 3]
```diff
@@ -47,10 +47,8 @@ cargo run --example csv_sql
 - [`advanced_udf.rs`](examples/advanced_udf.rs): Define and invoke a more complicated User Defined Scalar Function (UDF)
```
Comment (alamb) [inline datafusion-examples/README.md:47]: The removed examples were just inlined into the guide

File: docs/source/library-user-guide/using-the-sql-api.md [hunk 4]
```diff
@@ -19,4 +19,199 @@
...
+}
+```
+
+## Reading Multiple Files as a table
+
+It is also possible to read multiple files as a single table. This is done
+with the ListingTableProvider which takes a list of file paths and reads them
+as a single table, matching schemas as appropriate
+
 Coming Soon
```
Comment (alamb) [inline docs/source/library-user-guide/using-the-sql-api.md:166]: This would be cool to do but this PR is already large enought

**Review by jonahgao — APPROVED** [review]
Looks good to me!

**Comment by alamb** [comment]
Thank you for the review @jonahgao

**Pull request merged**